### ISSUE TITLE
About the finetunning, I want train my new dataset

### ISSUE THREAD

**Issue by wjy979769265** [minimaxir/gpt-2-simple#75]
There is no end sign in my dataset, in order to stop the generation for each sentence. I add start_token and end_token for each row, after finetunning, it generated sort of unreadable texts...It is because the BPE encoding or just the model treat the token as the normal text to train?
![image](https://user-images.githubusercontent.com/33248754/60176820-fb708280-9849-11e9-8a09-7a788b594c65.png)
After that, in order to achieve this function, I add a "." to each row, I got more readable text, but still not perfect, and I don't know how to decrease the loss... 
![image](https://user-images.githubusercontent.com/33248754/60176965-4e4a3a00-984a-11e9-86e6-724b7b1cb9ca.png)
Any one who have the idea please help me. Thank you very much.

**Comment by iedmrc** [comment]
Same here.
I have ~40k+ text files (~500mb) . 
What should be the best approach for data preprocessing? 
Should I add <|startoftext|> and <|endoftext|> tokens to the beginning and the end of files?
I use .npz encoder so what would be the best value for the `combine` parameter?
How do I decrease loss?

**Comment by iedmrc** [comment]
> Same here.
> I have ~40k+ text files (~500mb) .
> What should be the best approach for data preprocessing?
> Should I add` <|startoftext|>` and `<|endoftext|>` tokens to the beginning and the end of files?
> I use .npz encoder so what would be the best value for the `combine` parameter?
> How do I decrease loss?

I added `<|startoftext|>` and `<|endoftext|>` tokens to all of the files respectively. Seems it improved the accuracy (i mean, decreased the loss) a little bit. I didn't change the combine parameter because I thought `<|startoftext|>` and `<|endoftext|>` tokens did the job.
